crates/compiler/mono/src/ir.rs

```diff
@@ -9752,6 +10050,69 @@ fn from_can_pattern_help<'a>(
             })
         }
 
+        TupleDestructure {
+            whole_var,
+            destructs,
+            ..
+        } => {
+            // sorted fields based on the type
+            let sorted_elems = {
+                let mut layout_env = layout::Env::from_components(
+                    layout_cache,
+                    env.subs,
+                    env.arena,
+                    env.target_info,
+                );
+                crate::layout::sort_tuple_elems(&mut layout_env, *whole_var)
+                    .map_err(RuntimeError::from)?
+            };
+
+            // sorted fields based on the destruct
+            let mut mono_destructs = Vec::with_capacity_in(destructs.len(), env.arena);
+            let mut destructs_by_index = Vec::with_capacity_in(destructs.len(), env.arena);
+            destructs_by_index.extend(destructs.iter().map(Some));
+
+            let mut elem_layouts = Vec::with_capacity_in(sorted_elems.len(), env.arena);
+
+            // next we step through both sequences of elems. The outer loop is the sequence based
+            // on the type, since not all elems need to actually be destructured in the source
+            // language.
+            //
+            // However in mono patterns, we do destruct all patterns (but use Underscore) when
+            // in the source the elem is not matche in the source language.
```

Comment:
I don't really understand this comment. Is it still relevant.

Reply:
Copy-pasta. Yeah doesn't seem relevant to the current code. Gone.